### PR TITLE
Nanodet plus support

### PR DESCRIPTION
**Pull request by knoffelcut**
## Proposed change

This PR introduces support for [NanoDet-Plus](https://github.com/RangiLyu/nanodet). This series of models provides improvements over the default [SSDLite MobileNet](https://github.com/openvinotoolkit/open_model_zoo/tree/master/models/public/ssdlite_mobilenet_v2) model in both quality and footprint.

NanoDet-Plus-m-320: Comparable quality (COCO Map: 27.0 vs 24.3) at a lower footprint (FLOPS: 0.9G vs 1.525G) 
NanoDet-Plus-m-416: Improved quality (COCO Map: 30.4 vs 24.3) at the same footprint (FLOPS: 1.52G vs 1.525G) 

This model provided me with a better option in a power and cost constrained environment, e.g. detecting large objects using rpi3-like HW, using only the CPU

## Type of change

- [ ] Dependency upgrade
- [ ] Bugfix (non-breaking change which fixes an issue)
- [x] New feature
- [ ] Breaking change (fix/feature causing existing functionality to break)
- [ ] Code quality improvements to existing code
- [ ] Documentation Update

## Additional information

- This PR fixes or closes issue: None
- This PR is related to issue: None
- Link to discussion with maintainers (**required** for any large or "planned" features): None

## For new features

- [ ] There is an existing feature request or discussion with community interest for this change.
  - Link: 

## AI disclosure

- [x] No AI tools were used in this PR.

## Checklist

- [x] The code change is tested and works locally.
- [x] Local tests pass. **Your PR cannot be merged unless tests pass**
- [x] There is no commented out code in this PR.
- [x] I can explain every line of code in this PR if asked.
- [x] UI changes including text have used i18n keys and have been added to the `en` locale.
- [x] The code has been formatted using Ruff (`ruff format frigate`)

### CODE REVIEW

**Comment by hawkeye217** [comment]
What inference times are you seeing?

**Comment by knoffelcut** [comment]
I'll provide CPU inference time benchmarks on both ARM64 and AMD64 soon

**Comment by knoffelcut** [comment]
Results are in the tables below.

NanoDet-Plus does not does not compare well against SSDLite-MobileNet-V2 in SSDLite's best configuration. OpenVino does great with that SSDLite model.

I do not think it is worthwhile to use the pretrained NanoDet-Plus. The only case I can make is my use case, where I use it with custom trained data and a MobileNet-V4 backbone.

Thus, I have no qualms if you device to reject this PR

**One worthwhile finding is that is may be worthwhile to re-export the ssdlite_mobilenet_v2 model from OpenVino's Model Zoo using their updated tools, on the ARM64 platform it showed significant performance improvement.**

# Method
Measurements are the average over 128 samples. Before measuring 32 warmup iterations were performed.

The tables shows the inference framework with the source of model
| framework | description |
| - | - |
| ONNX FP32 | ONNX with FP32 precision |
| OpenVino FP32 (ONNX) | OpenVino with the ONNX model as input |
| OpenVino FP32 (Native) | OpenVino with the native (.xml/.bin) model as input, from the OpenVino model zoo |
| OpenVino FP16 (Native) | OpenVino with the native (.xml/.bin) model as input, from the OpenVino model zoo. FP16 variant[^1] |
| OpenVino FP16 (Frigate) | OpenVino with the native (.xml/.bin) model as input, from Frigate |

[^1]: I didn't expect this to be faster on CPU, but it allows for a direct comparison against the Frigate provided model

# Odroid-C4 (ARM64)
## Sample Duration [ms]
| Odroid-C4              | ONNX FP32 | OpenVino FP32 (ONNX) | OpenVino FP32 (Native) | OpenVino FP16 (Native) | OpenVino FP16 (Frigate) |
|-------------------------|-----------|----------------------|------------------------|------------------------|-------------------------|
| nanodet_plus_1x_m_320   | 114.64    | 86.27                |                        |                        |                         |
| nanodet_plus_1x_m_416   | 185.15    | 141.22               |                        |                        |                         |
| nanodet_plus_1.5x_m_416[^3] | 316       | 198.24               | 198.74                 | 196.38                 |                         |
| ssdlite_mobilenet_v2    | 219.25[^2]     | 143.61[^2]               | 72.42                  | 74.56                  | 75.25                   |

## Framerate [fps]
| Odroid-C4              | ONNX FP32 | OpenVino FP32 (ONNX) | OpenVino FP32 (Native) | OpenVino FP16 (Native) | OpenVino FP16 (Frigate) |
|-------------------------|-----------|----------------------|------------------------|------------------------|-------------------------|
| nanodet_plus_1x_m_320   | 8.7       | 11.6                 |                        |                        |                         |
| nanodet_plus_1x_m_416   | 5.4       | 7.1                  |                        |                        |                         |
| nanodet_plus_1.5x_m_416[^3] | 3.2       | 5.0                  | 5.0                    | 5.1                    |                         |
| ssdlite_mobilenet_v2    | 4.6[^2]       | 7.0[^2]                  | 13.8                   | 13.4                   | 13.3                    |

# x5950 (AMD64)
## Sample Duration [ms]
| x5950                   | ONNX FP32 | OpenVino FP32 (ONNX) | OpenVino FP32 (Native) | OpenVino FP16 (Native) | OpenVino FP16 (Frigate) |
|-------------------------|-----------|----------------------|------------------------|------------------------|-------------------------|
| nanodet_plus_1x_m_320   | 6         | 3.62                 |                        |                        |                         |
| nanodet_plus_1x_m_416   | 9.77      | 5.55                 |                        |                        |                         |
| nanodet_plus_1.5x_m_416[^3] | 11.77     | 6.98                 | 6.99                   | 6.93                   |                         |
| ssdlite_mobilenet_v2    | 10.6[^2]      | 13.11[^2]                | 2.47                   | 2.48                   | 2.94                    |

## Framerate [fps]
| x5950                   | ONNX FP32 | OpenVino FP32 (ONNX) | OpenVino FP32 (Native) | OpenVino FP16 (Native) | OpenVino FP16 (Frigate) |
|-------------------------|-----------|----------------------|------------------------|------------------------|-------------------------|
| nanodet_plus_1x_m_320   | 166.7     | 276.2                |                        |                        |                         |
| nanodet_plus_1x_m_416   | 102.4     | 180.2                |                        |                        |                         |
| nanodet_plus_1.5x_m_416[^3] | 85.0      | 143.3                | 143.1                  | 144.3                  |                         |
| ssdlite_mobilenet_v2    | 94.3[^2]      | 76.3[^2]                 | 404.9                  | 403.2                  | 340.1                   |

[^2]: I believe these are due to poor conversions from the tensorflow model to ONNX (using tf2onnx).
[^3]: From OpenVino model zoo

**Comment by hawkeye217** [comment]
Thanks for the benchmark numbers and the honest assessment.

Based on what you've discovered, I don't think we want to support a model architecture that's only marginally better when custom trained, the maintenance cost isn't justified when ssdlite under OpenVINO outperforms it for typical users. The note about re-exporting ssdlite with OpenVINO's updated tools is a good find, though, and is something that we'll investigate separately.

I'll close this, but thanks again for the contribution and the testing here.
